tfhe/src/integer/server_key/radix_parallel/tests_long_run/test_random_op_sequence.rs

```diff
@@ -0,0 +1,1195 @@
+        ),
+        (
+            Box::new(scalar_ne_executor),
+            &clear_ne,
+            "scalar ne".parse().unwrap(),
+        ),
+    ];
+
+    // Select Executor
+    let select_executor = CpuFunctionExecutor::new(&ServerKey::cmux_parallelized);
+
+    // Select
+    let clear_select = |b: bool, x: u64, y: u64| if b { x } else { y };
+
+    #[allow(clippy::type_complexity)]
+    let mut select_op: Vec<(SelectOpExecutor, &dyn Fn(bool, u64, u64) -> u64, String)> = vec![(
+        Box::new(select_executor),
+        &clear_select,
+        "select".parse().unwrap(),
+    )];
+
+    // Div executor
+    let div_rem_executor = CpuFunctionExecutor::new(&ServerKey::div_rem_parallelized);
+    // Div Rem Clear functions
+    let clear_div_rem = |x: u64, y: u64| -> (u64, u64) { (x / y, x % y) };
+    #[allow(clippy::type_complexity)]
+    let mut div_rem_op: Vec<(DivRemOpExecutor, &dyn Fn(u64, u64) -> (u64, u64), String)> = vec![(
+        Box::new(div_rem_executor),
+        &clear_div_rem,
+        "div rem".parse().unwrap(),
+    )];
+
+    // Scalar Div executor
+    let scalar_div_rem_executor = CpuFunctionExecutor::new(&ServerKey::scalar_div_rem_parallelized);
+    #[allow(clippy::type_complexity)]
+    let mut scalar_div_rem_op: Vec<(
+        ScalarDivRemOpExecutor,
+        &dyn Fn(u64, u64) -> (u64, u64),
+        String,
+    )> = vec![(
+        Box::new(scalar_div_rem_executor),
+        &clear_div_rem,
+        "scalar div rem".parse().unwrap(),
+    )];
+
+    // Log2/Hamming weight ops
+    let ilog2_executor = CpuFunctionExecutor::new(&ServerKey::ilog2_parallelized);
+    let count_zeros_executor = CpuFunctionExecutor::new(&ServerKey::count_zeros_parallelized);
+    let count_ones_executor = CpuFunctionExecutor::new(&ServerKey::count_ones_parallelized);
+    let clear_ilog2 = |x: u64| x.ilog2() as u64;
+    let clear_count_zeros = |x: u64| x.count_zeros() as u64;
+    let clear_count_ones = |x: u64| x.count_ones() as u64;
+
+    #[allow(clippy::type_complexity)]
+    let mut log2_ops: Vec<(Log2OpExecutor, &dyn Fn(u64) -> u64, String)> = vec![
+        (
+            Box::new(ilog2_executor),
+            &clear_ilog2,
+            "ilog2".parse().unwrap(),
+        ),
+        (
+            Box::new(count_zeros_executor),
+            &clear_count_zeros,
+            "count zeros".parse().unwrap(),
+        ),
+        (
+            Box::new(count_ones_executor),
+            &clear_count_ones,
+            "count ones".parse().unwrap(),
+        ),
+    ];
+
+    random_op_sequence_test(
+        param,
+        &mut binary_ops,
+        &mut unary_ops,
+        &mut scalar_binary_ops,
+        &mut overflowing_ops,
+        &mut scalar_overflowing_ops,
+        &mut comparison_ops,
+        &mut scalar_comparison_ops,
+        &mut select_op,
+        &mut div_rem_op,
+        &mut scalar_div_rem_op,
+        &mut log2_ops,
+    );
+}
+
+#[allow(clippy::too_many_arguments)]
+pub(crate) fn random_op_sequence_test<P>(
```

Comment:
One suggestion to simplify a bit checks in the if and the subtraction done to get the index:

```rust
    let binary_ops_range = 0..binary_ops.len();
    let unary_ops_range = binary_ops_range.end..binary_ops_range.len()+unary_ops.len();
    // ..etc

    let index = 0;
    if binary_ops_range.contains(&index) {
        let i = index - binary_ops_range.start;
    } else if unary_ops_range.contains(&index) {
        let i = index - unary_ops_range.start;
    } // etc...
```

or, pick the thin using 2 randoms: one to get the op type, then another to get the op


```rust
let op_type_index = gen.range(0..num_op_type);
match op_type_index {
0 => { 
    let op_index = en.range(0..binary_ops.len());
   /...
}
1 => { 
    let op_index = en.range(0..unary_ops.len());
   /...
}
// ..
_ => panic!("invalid op type index")
```